sidekiq/queue.go

```diff
@@ -137,8 +138,13 @@ func NewQueue(client redis.UniversalClient) Queue {
 	if sidekiq_ns ~= "" then
 		queue_key = table.concat({sidekiq_ns, queue_key}, ":")
 	end
-	if redis.call("exists", queue_key) == 1 then
-		redis.call("rename", queue_key, puller_queue_key)
+
+	for i = 1,max_jobs do
+		local jobm = redis.call("rpop", queue_key)
+		if jobm == false then
+			break
+		end
+		redis.call("lpush", puller_queue_key, jobm)
```

Comment:
I'd suggest adding a test for this functionality. Also, will redis just create the `puller_queue_key` list when you call `lpush` if it doesn't already exist?

Reply:
Yes, that is tested